src/gui/tray/Window.qml

```diff
@@ -624,56 +616,156 @@ Window {
                     }
-                    ToolTip.visible: hovered
-                    ToolTip.delay: 1000
-                    ToolTip.text: qsTr("Open Talk")
-                    onClicked: Qt.openUrlExternally(link)
 
-                    Accessible.role: Accessible.Button
-                    Accessible.name: qsTr("Open Talk %1").arg(link)
-                    Accessible.onPressAction: replyButton.clicked()
+                    Button {
+                        id: moreActionsButton
+
+                        Layout.preferredWidth: parent.height
+                        Layout.preferredHeight: parent.height
+                        Layout.alignment: Qt.AlignRight
+
+                        flat: true
+                        hoverEnabled: true
+                        visible: activityItem.links.length > 2
+                        display: AbstractButton.IconOnly
+                        icon.source: "qrc:///client/theme/more.svg"
+                        icon.color: "transparent"
+                        background: Rectangle {
+                            color: parent.hovered ? Style.lightHover : "transparent"
+                        }
+                        ToolTip.visible: hovered
+                        ToolTip.delay: 1000
+                        ToolTip.text: qsTr("Show all actions")
+                        onClicked:  moreActionsButtonContextMenu.popup();
+
+                        Connections {
+                            target: trayWindow
+                            onActiveChanged: {
+                                if (!trayWindow.active && moreActionsButtonContextMenu.opened) {
+                                    moreActionsButtonContextMenu.close();
+                                }
+                            }
+                        }
+
+                        Connections {
+                            target: activityListView
+
+                            onMovementStarted: {
+                                if (moreActionsButtonContextMenu.opened) {
+                                    moreActionsButtonContextMenu.close();
+                                }
+                            }
+                        }
+
+                        Accessible.role: Accessible.Button
+                        Accessible.name: qsTr("Show all actions")
+                        Accessible.onPressAction: moreActionsButton.clicked()
+
+                        Container {
+                            id: moreActionsButtonContextMenuContainer
+                            visible: moreActionsButtonContextMenu.opened
+                            Menu {
+                                id: moreActionsButtonContextMenu
+                                anchors.centerIn: parent
+
+                                onOpenedChanged: {
+                                    if (opened) {
+                                        moreActionsButtonContextMenuContainer.width = moreActionsButtonContextMenu.width;

```

Comment:
Why isn't this done with bindings?

Also that feels like using `childrenRect` on the moreActionsButtonContextMenuContainer but I'm not sure.
https://doc.qt.io/qt-5/qml-qtquick-item.html#childrenRect-prop